File: src/algorithms/smallest-equal.js
/**
 * Given an array nums, returns the smallest index of nums such that
 * index % 10 === nums[index]. Returns -1 if no such index exists.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number} - Returns an index
 */
function smallestEqual (nums) {
  for (let i = 0; i < nums.length; i++) {
    if (i % 10 === nums[i]) {
      return i;
    }
  }
  return -1;
}

File: src/algorithms/smallest-equal.py
def smallestEqual(nums):
    """Given a 0-indexed integer array nums, return the smallest index i of 
    nums such that i mod 10 == nums[i], or -1 if such index does not exist. x 
    mod y denotes the remainder when x is divided by y.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns an index.
    """
    for i in range(len(nums)):
        if i % 10 == nums[i]:
            return i
    return -1
